### PR TITLE
Add a script to run iOS test app on AWS Device Farm

### PR DESCRIPTION
**Pull request by huydhn**
This adds a script to test PyTorch on actual iOS devices on AWS Device Farm. The test could take quite a long time pending for the devices to become available, so the steps are done manually and documented in `ios/TestApp/README.md`.

### Testing

1. TestApp itself runs fine on my local iPhone 13 and on [device farm](https://us-west-2.console.aws.amazon.com/devicefarm/home#/mobile/projects/b531574a-fb82-40ae-b687-8f0b81341ae0/runs/d2653ca8-8ee2-44dd-b15e-0402f9ab0aca).  I can see the benchmark results output at the console log.
```
BUILD_LITE_INTERPRETER=1 USE_PYTORCH_METAL=1 USE_COREML_DELEGATE=1 IOS_PLATFORM=OS IOS_ARCH=arm64 ./scripts/build_ios.sh

pushd ios/TestApp/benchmark
ruby setup.rb --lite 1 -t 9HKVT38N77 --benchmark
popd

ruby scripts/xcode_build.rb -i build_ios/install -x ios/TestApp/TestApp.xcodeproj -p "OS"
```

2. Trying to run TestAppTests https://github.com/pytorch/pytorch/blob/main/ios/TestApp/TestAppTests/TestLiteInterpreter.mm on my local iPhone ends up with this error `Logic Testing Unavailable. Logic Testing on iOS devices is not supported. You can run logic tests on the Simulator`.  I update the xcode project to reuse TestApp as the host application.
```
ruby setup.rb --lite 1 -t 9HKVT38N77
```

3.. Trying [another round of testing on device farm](https://us-west-2.console.aws.amazon.com/devicefarm/home#/mobile/projects/b531574a-fb82-40ae-b687-8f0b81341ae0/runs/18dbd69d-8608-46d8-a868-bd05b69375db)

### CODE REVIEW

**Comment by huydhn** [comment]
Screenshot of the benchmark test on my iphone, it runs some inference rounds with lite interpreter:

IMG_0556

**Comment by huydhn** [comment]
https://us-west-2.console.aws.amazon.com/devicefarm/home#/mobile/projects/b531574a-fb82-40ae-b687-8f0b81341ae0/runs/18dbd69d-8608-46d8-a868-bd05b69375db fails one testCoreML test on iPhone 13, it looks like a flaky issue though as the test passes everywhere else, so I'm retrying it.

File: ios/TestApp/README.md [hunk 1]
```diff
@@ -44,6 +44,85 @@ python test/mobile/model_test/gen_test_model.py ios-test
 cd ios/TestApp/benchmark; python coreml_backend.py
 ```
 
+## Run test on AWS Device Farm
+The test app and its test suite could also be run on actual devices via
+AWS Device Farm.
+
+1. Checkout PyTorch repo including all submodules
+
+2. Build PyTorch for iOS devices, not for simulator
```
Comment (kit1980) [inline ios/TestApp/README.md:53]: Do you need to do this on MacOS? Do you need to have XCode installed?

**Review by kit1980 — APPROVED** [review]

File: ios/TestApp/README.md (hunk 1 above)
Comment (huydhn) [reply ios/TestApp/README.md:53]: Oh yeah, you need to run this on MacOS with XCode installed.  Let me clarify this as a prerequisite

**Pull request closed without merging**